languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -9779,6 +9779,33 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+    <rulegroup id="RISQUE_ET_CHANCE" name="risque et chance">
+      <rule>
+          <pattern>
+              <marker><token inflected="yes">risquer</token></marker>
+              <token regexp="yes">de</token>
+              <token regexp="yes">gagner</token>
+          </pattern>
+          <message>Faute de style possible car cette expression peut paraître péjorative.
+              Nous vous suggérons <suggestion>avoir des chances</suggestion>.</message>
```

Comment:
No line breaks allowed in `<message>`s :)

Also, it would be best if we proposed the verb "avoir" in the same form the verb "risquer" is in the original. Otherwise, if the user writes "Il risque de gagner", our correction will result in "Il avoir des chances de ganger" (unacceptable).

Unfortunately, this is a little complicated due to "avoir" having "avoir" in the postag, contrary to "normal" verbs. This is how it can be done:

```xml
      <rule>
          <pattern>
              <marker>
                <token postag="V (ind|sub).*" postag_regexp="yes" inflected="yes">risquer</token>
              </marker>
                <token regexp="yes">de</token>
                <token regexp="yes">gagner</token>
          </pattern>
          <message>Faute de style possible car cette expression peut paraître péjorative.</message>
          <suggestion><match no="1" postag="(V )(ind|sub)(.*)" postag_regexp="yes" postag_replace="(V avoir ind$3)|(V avoir sub$3)">avoir</match> des chances</suggestion>
          <example correction="a des chances|ait des chances">Il <marker>risque</marker> de gagner.</example>
      </rule>
```

About formatting: please notice that we normally don't put `<marker><token/></marker>` in one line, but in three separate lines.